### ISSUE TITLE
Linux/MacOS Pixelated, Low Resolution

### ISSUE THREAD

**Issue by ryanwwest** [koreader/koreader#10550]
* KOReader version: 2023.03-25
* Device: Macbook Pro 2021 16 inch, Surface Pro 9 (WSL)

#### Issue

On both Linux (WSL) and MacOS, KOReader runs at a lower resolution compared to the rest of my display/windows. You can see this in [this issue](https://github.com/koreader/koreader/discussions/10372) for Linux and for Mac (compare the window pixelation compared to the window titlebar and other surrounding text):

![image](https://github.com/koreader/koreader/assets/31378238/e35ae9f1-d34e-4139-b846-f49c414ade93)

Gear -> Screen -> Screen DPI doesn't affect this, it seems to only scale the size of the UI elements (of which the minimum, 90, is still slightly high for big screens).

Is there any way to use the full resolution? I can't find anything about how to fix it, so it seems like a bug.

**Comment by Frenzie** [comment]
> Is there any way to use the full resolution?

The short answer, tell macOS/Wayland not to be astoundingly idiotic with "HiDPI" and to support it properly the way e.g. Windows XP and X11 do.

> 2023.03-25

The longer answer, that's your problem right there. ;-)

**Comment by ryanwwest** [comment]
Thanks, did not realize a fix was already made so recently. Though a sampling of the most recent MacOS Github build artifacts seems to be keeping the app from fully starting due to a dynamic library loading issue:

```
ffi.load: libs/libmupdf.dylib
ffi.load (warning): dlopen(libs/libmupdf.dylib, 0x0005): Library not loaded: '@rpath/libsharpyuv.0.dylib'
  Referenced from: '/Users/rw/Downloads/KOReader.app/Contents/koreader/libs/libwebp.7.dylib'
  Reason: tried: '/Users/rw/Downloads/KOReader.app/Contents/koreader/libs/../lib/libsharpyuv.0.dylib' (no such file), '$ORIGIN/libsharpyuv.0.dylib' (no such file), '$ORIGIN/libsharpyuv.0.dylib' (no such file), '/usr/local/lib/libsharpyuv.0.dylib' (no such file), '/usr/lib/libsharpyuv.0.dylib' (no such file)
failed to run lua chunk: ./setupkoenv.lua:27: Not able to load dynamic library: libs/libmupdf.dylib
```

Interestingly the path `/Users/rw/Downloads/KOReader.app/Contents/koreader/libs/../lib/libsharpyuv.0.dylib` doesn't exist because of the `../lib/` part - the library shows up within `/libs` so maybe this is handled by some ko environment source script. But I'm not sure why it only isn't working now, and works fine with the 2023.03-25 version.

**Comment by ryanwwest** [comment]
It works on latest update! Looks like `libs/` in the macos package indeed needs to be renamed to `lib/`, which prevents the app from starting until doing so, so that might be a bug. But the one I opened this issue for, is working on latest 2023.06.

**Comment by melyux** [comment]
Latest artifact is still not working for me. When I renamed `libs/` to `lib/` the other libraries stopped working. So I made a separate `lib/` and put `libsharpyuv.0.dylib` in there.

This got further, but ultimately failed. The app crashes immediately on start with `zsh: bus error  /Applications/KOReader.app/Contents/MacOS/koreader`.

**Comment by benoit-pierre** [comment]
I would be really interested to know if my [meson macos builds](https://github.com/benoit-pierre/koreader/actions/runs/5523187873) work (or more like how broken they are!). I don't have a machine to test. I tried with [darling](https://github.com/darlinghq/darling), but get an error during the SDL init. I can get most of the testsuite to pass on the CI, so I think I got at least some parts right.
